src/index.ts

```diff
@@ -183,3 +193,66 @@ function encodeCursor<Record, Cursor, Node, CustomEdge extends Edge<Node>>(
 ): string {
   return options.encodeCursor(options.getCursor(record))
 }
+
+
+export async function findManyCursorConnectionWithPageCursors<
+  Record = { id: string },
+  Cursor = { id: string, page: number },
+  Node = Record,
+  CustomEdge extends Edge<Node> = Edge<Node>
+>(
+  findMany: (args: PrismaFindManyArguments<Cursor>) => Promise<Record[]>,
+  aggregate: () => Promise<number>,
+  args: ConnectionArguments = {},
+  pOptions?: Omit<Options<Record, Cursor, Node, CustomEdge>, "decodeCursor" | "encodeCursor" | "findManyParamsWithCursor">
+): Promise<Connection<Node, CustomEdge>> {
+
+  const options: Options<Record, Cursor, Node, CustomEdge> = {
+    ...pOptions,
+    decodeCursor: (cursor: string) => {
+      // All cuids start with "c", so if it's not then we're good to go
+      if (!cursor.startsWith("c")) {
+        const [page, id] = cursor.split("-")
+        return ({ id, page: parseInt(page) } as unknown as Cursor)
+      } else {
+        return ({ id: cursor, page: 0 } as unknown as Cursor)
+      }
+    },
```

Comment:
constraint 1: people need to have objects with an id that is cuid (this is something everyone probably needs for using relay object identification

Reply:
I have been thinking - would this be possible without this constraint as well? I know some folks (me included) prefer database-native types (i.e. PostgreSQL's UUID type) over CUIDs.

From how I understand this code from a first skim - and feel free to correct me here - the only real requirement on CUIDs is this line

```ts
const firstItem = await findMany({ take: 1, orderBy: { id: direction } })
```

How would this behave if the user already provided their own `orderBy` (e.g. by modification date)? And could we just require the user to provide an `orderBy` when they call `findManyCursorConnectionWithPageCursors`, probably via an option?

Reply:
The constraint comes from being able to guarantee that you have a cuid, and not the page-info'd version because it starts [with `c`](https://github.com/ericelliott/cuid#broken-down) in `if (!cursor.startsWith('c')) {`. With an opaque db type we'd need some other technique for knowing whether it's the pageInfo+cursor encoded cursor or not. Perhaps a forced unbase64-ing and then checking if that contained a specific encoded delimintator could work but it's not guaranteed to work - open to ideas though

Reply:
Since we have full control over encoding and decoding cursor, and it's supposed to be opaque to the client with the only requirement being that the output is a string, we could encode any information into it, e.g. via a base64'd JSON ala `{ i: '<id>', p: 1 }` for the page-info'd one, and `{ i: '<id>' }` for the basic cursor.

It could even support the usual `getCursor` that is described [here](https://github.com/devoxa/prisma-relay-cursor-connection#custom-cursors) and just tuck on the page information.